cads_adaptors/adaptors/arco.py

```diff
@@ -64,14 +90,21 @@ def _normalise_data_format(self, request: Request) -> None:
             f"Invalid {data_format=}. Available options: {available_options}"
         )
 
+    def pre_mapping_modifications(self, request: Request) -> Request:
+        request = super().pre_mapping_modifications(request)
+
+        download_format = request.pop("download_format", "as_source")
+        self.set_download_format(download_format)
+        return request
+
     def normalise_request(self, request: Request) -> Request:
         if self.normalised:
             return request
 
         request = copy.deepcopy(request)
         self._normalise_variable(request)
         self._normalise_location(request)
-        self._normalise_date(request)
+        self._normalise_date(request, date_key=self.config.get("date_key", "date"))
```

Comment:
Why is `date_key` a class method argument inherited from self?
Can you introduce a new property?
```python
@property
def date_key(self) -> str:
    return self.config.get("date_key", "date")
```

Reply:
Yes, but not required.